vividus-extension-selenium/src/main/java/org/vividus/steps/ui/BarcodeSteps.java

```diff
@@ -69,23 +77,38 @@ public BarcodeSteps(ScreenshotTaker screenshotTaker, BarcodeActions barcodeActio
      * <li><b>NEXT_BATCHES</b> - the variable will be available starting from next batch
      * </ul>
      * @param variableName A name under which the value should be saved
-     * @throws IOException If an input or output exception occurred
      */
     @When("I scan barcode from screen and save result to $scopes variable `$variableName`")
```

Comment:
still there should be 2 steps:
- the following step should work as it worked before (viewport screenshot): 
    ```
    When I scan barcode from screen and save result to $scopes variable `$variableName`
    ```
- the new step should be implemented and it should follow contextual approach: 
    ```
    When I scan barcode from context and save result to $scopes variable `$variableName`
    ```

Reply:
from screen step should be deprecated?

Reply:
nope, as we don't have full-page screenshots for mobile native apps